### PR TITLE
BKNDLSS-23694 - [SUPPORT] Find data objects via POST

### PR DESCRIPTION
**Pull request by viktor-liablin**
BKNDLSS-23694 - [SUPPORT] Find data objects via POST

### CODE REVIEW

File: test/unit/specs/data/find.js [hunk 1]
```diff
@@ -28,10 +28,10 @@ describe('<Data> Find', function() {
       const result1 = await dataStore.find()
```
Comment (Valodya) [inline test/unit/specs/data/find.js:28]: Is there a test for running `dataStore.find()` with all the available parameters?

Comment (viktor-liablin) [reply test/unit/specs/data/find.js:28]: No. I've tested them manually. I will create them

Comment (Valodya) [reply test/unit/specs/data/find.js:28]: ok, there should both `unit` and `e2e` tests

File: src/data/store.js [hunk 2]
```diff
@@ -24,6 +24,20 @@ function buildFindFirstLastQuery(queryBuilder, sortDir) {
   return DataQueryBuilder.toRequestBody(query)
 }
 
+function buildFindQuery(queryBuilder) {
```
Comment (Valodya) [inline src/data/store.js:27]: this logic is already persisted in the `DataQueryBuilder.toRequestBody` method

Comment (viktor-liablin) [reply src/data/store.js]: \+

File: test/unit/specs/data/find.js (hunk 1 above)
Comment (viktor-liablin) [reply test/unit/specs/data/find.js:28]: \+

File: test/e2e/helpers/utils.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    hugeRequest += addToRequest
+    cloneCount++
+    currentCharts += addToRequest.length
+  }
+
+  return hugeRequest += `${columnName} != 'hugerequest${cloneCount}'`
+}
+
+export function sortByProperty(property) {
```
Comment (Valodya) [inline test/e2e/helpers/utils.js:17]: `sortByProperty` => `createComparator`

File: test/e2e/helpers/utils.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+export function getHugeMySQL(minCharts, columnName) {
```
Comment (Valodya) [inline test/e2e/helpers/utils.js:1]: seems like this is needed for a single case, move to the file where it is used

Comment (Valodya) [reply test/e2e/helpers/utils.js:1]:
```
function generateLongString(length = 1000){
   return new Array(length).fill('a').join('')
}
```

File: test/e2e/specs/data-where-clause.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      result.sort(sortByProperty('name'))
+    }
+
+    expect(result).to.eql(expected)
+  })
+
+  it('should work with huge query', async () => {
+    let hugeFakeMySQL = getHugeMySQL(2048, 'name')
+
+    hugeFakeMySQL = `${hugeFakeMySQL} AND name != 'name1' AND name != 'name2'`
```
Comment (Valodya) [inline test/e2e/specs/data-where-clause.js:115]:
```
const where = `name = '${generateLongString(3000)}'`
```

File: test/e2e/helpers/utils.js (hunk 3 above)
Comment (viktor-liablin) [reply test/e2e/helpers/utils.js]: \+

File: test/e2e/helpers/utils.js (hunk 4 above)
Comment (viktor-liablin) [reply test/e2e/helpers/utils.js]: \+

File: test/e2e/specs/data-where-clause.js (hunk 5 above)
Comment (viktor-liablin) [reply test/e2e/specs/data-where-clause.js]: \+

**Pull request merged**